<a href="https://colab.research.google.com/github/ilinthm26/Gender-Inclusivity-and-Pay-Gap-Analysis/blob/main/IBM_Gender_Inclusivity_and_Pay_Gap_Statistical_Test.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ***IBM Gender Inclusivity and Pay Gap Statistical Test***

# Import Libraries and Load Data

In [1]:
import pandas as pd
import numpy as np
from scipy import stats

In [4]:
df=pd.read_csv(r'/content/processed_hr_data_eda.csv')

# Prepare Data

In [6]:
male_salary = df[df['Gender']=='Male']['MonthlyIncome']
female_salary=df[df['Gender']=='Female']['MonthlyIncome']

print("Male Sample Size:",len(male_salary))
print("Female Sample Size:",len(female_salary))

Male Sample Size: 882
Female Sample Size: 588


## T-Test (Welch's t-test)



In [7]:
t_stat,p_value=stats.ttest_ind(
    male_salary,
    female_salary,
    equal_var=False #safer when variance differ
)

print("T-Statistics:",round(t_stat,4))
print("P-Value:",round(p_value,6))

T-Statistics: -1.2223
P-Value: 0.221837


Hypothesis Testing Result:

In [9]:
alpha=0.05

if p_value < alpha:
    print("✅ p < 0.05 → Reject H0")
    print("Conclusion: The salary difference is statistically significant.")
else:
    print("❌ p ≥ 0.05 → Fail to Reject H0")
    print("Conclusion: No statistically significant salary difference detected.")

❌ p ≥ 0.05 → Fail to Reject H0
Conclusion: No statistically significant salary difference detected.


# Effects Size / Cohen's d
| Cohen’s d | Meaning      |
| --------- | ------------ |
| 0.2       | Small effect |
| 0.5       | Medium       |
| 0.8       | Large        |



In [10]:
mean_diff=male_salary.mean()-female_salary.mean()
pooled_std=np.sqrt(
    ((male_salary.std()**2)+(female_salary.std()**2))/2
)
cohen_d=mean_diff/pooled_std

round(cohen_d,3)

np.float64(-0.065)



An independent Welch’s t-test was conducted to evaluate whether salary differences exist between male and female employees.

The test produced a t-statistic of -1.22 and a p-value of 0.22.

Since the p-value exceeds the 0.05 threshold, we fail to reject the null hypothesis. This indicates that the observed salary difference is not statistically significant.

Furthermore, the effect size (Cohen’s d = -0.065) suggests the practical difference between genders is negligible.

Overall, the analysis does not provide evidence of a meaningful gender-based salary disparity within this dataset.
